### PR TITLE
Path TryInto String

### PR DESCRIPTION
**Pull request by mattyg**
### Summary
Adds TryInto<String> for path for easy conversion of paths to a string representation

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/hdk/src/hash_path/path.rs [hunk 1]
```diff
@@ -578,4 +598,7 @@ fn hash_path_path() {
     ] {
         assert_eq!(Path::from(input), Path::from(output),);
     }
+
+    let path_to_string: String = Path::from("foo.a.b.c.abcdef.bar").try_into().unwrap();
+    assert_eq!(String::from("foo.a.b.c.abcdef.bar"), path_to_string,);
```
Comment (jost-s) [inline crates/hdk/src/hash_path/path.rs:603]:
```suggestion
    let path = "foo.a.b.c.abcdef.bar";
    let path_to_string: String = Path::from(path).try_into().unwrap();
    assert_eq!(path.to_string(), path_to_string,);
```

**Review by jost-s — APPROVED** [review]
LGTM

Do us the honor of adding this change to the HDK changelog, @mattyg =)

**Review by ThetaSinner — APPROVED** [review]

**Comment by mattyg** [comment]
> LGTM
> 
> Do us the honor of adding this change to the HDK changelog, @mattyg =)

gladly! :laughing:

**Review by maackle — APPROVED** [review]

**Comment by ThetaSinner** [comment]
@mattyg I believe this will build for you now if you rebase. We aren't able to get the PR workflow to run for forks for now in a secure way. I've added you as a contributor for Holochain so you'll have the access you need.

If you're interested why, we have self-hosted runners for Github actions and those have a global login to Cachix (all this is public anyway in another repo so nothing secret). So unless we can get to a point where Holochain's tests can be sandboxed without public internet access it's not going to be safe to run code from forked PRs on those runners. We're looking at making environment changes to support scale testing and hopefully the infrastructure changes we make as part of that can trickle down to CI and solve this problem 🤞

**Comment by mattyg** [comment]
closing this, and reopening as #2544 with branch inside holochain repo so CI can run

**Pull request closed without merging**